src/components/PublicMentorPage/PublicMentorPage.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { type FC, useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import MentorCard from '../MentorCard/MentorCard.component';
+import { type MentorCardType } from '../../types';
+import { API_URL } from '../../constants';
+
+const PublicMentorPage: FC = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const queryParams = new URLSearchParams(location.search);
+  const categoryParam = queryParams.get('category');
+
+  const [mentors, setMentors] = useState<MentorCardType[]>([]);
+  const [categories, setCategories] = useState<string[]>([]);
+  const [selectedCategory, setSelectedCategory] = useState<string | null>(
+    categoryParam
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    const mentorsUrl = `${API_URL}/mentors`;
+    const categoriesUrl = `${API_URL}/categories`;
+
+    Promise.all([
+      fetch(mentorsUrl).then(async (response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return await response.json();
+      }),
```

Comment:
Let's create a custom hook for this so we can reuse this. Refer to the useCategory hook